core/capstanignore.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package core
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Capstanignore interface {
+	LoadFile(path string) error
+	AddPattern(pattern string) error
+	PrintPatterns()
+	IsIgnored(path string) bool
+}
+
+var CAPSTANIGNORE_ALWAYS []string = []string{"/meta/*", "/mpm-pkg/*", "/.git/*"}
+
+// CapstanignoreInit creates a new capstanignore struct that is
+// used when deciding whether a file should be included in unikernel
+// or not. You can provide `path` to the .capstanignore file to load
+// it or pass empty string "" if you have none. Note that once having
+// capstanignore struct you can load as many files as you want (using
+// .LoadFile function) or manually add as many patterns as you like
+// (using .AddPattern function).
+func CapstanignoreInit(path string, verbose bool) Capstanignore {
+	c := capstanignore{verbose: verbose}
+
+	// Load capstanignore file if path was given.
+	if path != "" {
+		if err := c.LoadFile(path); verbose && err != nil {
+			fmt.Println("WARN: failed to load .capstanignore file:", err)
+		}
+	}
+
+	// Always ignore some common paths.
+	for _, pattern := range CAPSTANIGNORE_ALWAYS {
+		c.AddPattern(pattern)
+	}
+	return &c
+}
+
+type capstanignore struct {
+	ignored  []string         // list of all ignored patterns
+	ignoredC []*regexp.Regexp // list of compiled patterns
```

Comment:
Since you are only supporting ignore patterns, wouldn't it be fine if you named these two as `patterns` and `compiledPatterns`? The `ignoreC` is strangely compact.